src/Model/Lumberjack.php

```diff
@@ -113,8 +113,9 @@ protected function excludeSiteTreeClassNames($list)
         $classNames = $this->owner->getExcludedSiteTreeClassNames();
         if ($this->shouldFilter() && count($classNames)) {
             // Filter the SiteTree
-            $list->exclude('ClassName', $classNames);
+            $list = $list->exclude('ClassName', $classNames);
         }
+        return $list;
```

Comment:
Nice catch, `DataList::exclude` returns a new list rather than modifying the original. Would you mind updating the doc block for this method to make it clear that it returns a new DataList as well?

Reply:
No problem! My only fear is that you're now going to ask me to squash those two commits into one (and I have to go digging to find out how, again...) ;-) Just let me know if you do.

Reply:
😄  we can do that on merge

Reply:
Awesome - Is this thread's request for a comment not the same as the other request (ie: "updating the doc block..." vs: "message here to say that it..."?

Reply:
Yeah, it's the same request

Reply:
No problem, well then we're all set, you have the patch with the two commits. Thanks @robbieaverill

Reply:
Sorry @axllent, can you please change the doc block to indicate that it returns a new instance - here's an example:

```php
/**
 * Excludes any hidden owner subclasses. Note that the returned DataList will be a different
 * instance from the original
 *
 * @param DataList $list
 * @return DataList
 */
```

Reply:
No problem, done @robbieaverill

Reply:
Arrg, hang one ... need an extra line in there... sorry, bare with me.

Reply:
Okidoki, better now. It's a good thing you can squash ;-)